frontend/src/types/userCart.ts

```diff
@@ -0,0 +1,53 @@
+import { Donut } from "../../database/schemas/donut_schema"
+
+
+export interface UserCart {
+  "donuts": Donut[], // List of ordered doughnuts
+  "quantities": number[], // Parallel list to doughnuts with the number of doughnuts being bought.
+}
+
+export type CartReducerAction = {
+  name: 'set' | 'clear';
```

Comment:
enum might be a bit cleaner than two strings

Reply:
I'm not sure. It feels like extra overhead since typescript checks anyways. Not going to change